information/specification.md

```diff
@@ -0,0 +1,826 @@
+    This failure occurs when the total value of the entries within the [initial
+    UTxO set](#initial-utxo-set) (the amount of money _available_) is _less
+    than_ the the total value of all entries in the [requested output
+    list](#requested-output-list) (the amount of money _required_).
+
+  * #### UTxO Not Fragmented Enough
+
+    This failure occurs when the _number_ of entries in the [initial UTxO
+    set](#initial-utxo-set) is _smaller than_ the number of entries in the
+    [requested output list](#requested-output-list), for algorithms that impose
+    the restriction that a single UTxO entry can only be used to pay for _at
+    most one_ output.
+
+  * #### UTxO Fully Depleted
+
+    This failure occurs if the algorithm depletes all entries from the [initial
+    UTxO set](#initial-utxo-set) _before_ it is able to pay for all outputs in
+    the [requested output list](#requested-output-list).
+
+    This can happen _even if_ the total value of entries within the [initial
+    UTxO set](#initial-utxo-set) is _greater than_ the total value of all
+    entries in the [requested output list](#requested-output-list), due to
+    various restrictions that coin selection algorithms impose on themselves
+    when selecting UTxO entries.
+
+  * #### Maximum Input Count Exceeded
+
+    This failure occurs if the _number_ of UTxO entries needed to pay for the
+    outputs in the [requested output list](#requested-output-list) exceeds the
+    upper limit specified by the [maximum input count](#maximum-input-count)
+    parameter.
+
+# Implementations
+
+This section will describe the coin selection algorithms used by Cardano Wallet.
+
+These algorithms implement a [common interface](#interface), as described
+above.
+
+There are two main algorithms used by Cardano Wallet:
+
+  * [Largest-First](#largest-first)
+  * [Random-Improve](#random-improve)
+
+In general, Cardano Wallet gives _priority_ to the
+[Random-Improve](#random-improve) algorithm, as experimental evidence shows
+that it performs better at [minimising dust](#goals) and maintaining a UTxO set
+with [useful outputs](#goals). (See [Self Organization in Coin
+Selection](#self-organization-in-coin-selection) for more details.)
+
+However, in rare cases, the [Random-Improve](#random-improve) may fail to
+produce a result. In such cases, Cardano Wallet will fall back to the
+[Largest-First](#largest-first) algorithm.
+
+## Largest-First
+
+The **Largest-First** coin selection algorithm processes
+[outputs](#requested-output-list) in _descending order of coin value_, from
+_largest_ to _smallest_.
+
+For each output, it repeatedly selects the _largest remaining_ unspent UTxO
+entry until the value of selected entries is greater than or equal to the
+value of that output.
+
+The name of the algorithm is taken from the idea that the **largest** UTxO
+entry is always considered **first**.
+
+### Cardinality
+
+The Largest-First algorithm imposes the following cardinality restriction:
```

Comment:
See my previous comment, this restriction should only apply to the Random-Improve algorithm!

Reply:
Fixed in d35f56c06f8924c403b915317e967ea84d212801.